packages/webdriverio/src/scripts/shadowFnFactory.js

```diff
@@ -0,0 +1,13 @@
+// generate a function that can be used to query shadowRoots
+export const shadowFnFactory = function(elementSelector, qsAll) {
+    const strFn = `
+    (function() {
+      // element has a shadowRoot property
+      if (this.shadowRoot) {
+        return this.shadowRoot.querySelector${qsAll ? 'All' : ''}('${elementSelector}')
+      }
+      // fall back to querying the element directly if not
+      return this.querySelector${qsAll ? 'All' : ''}('${elementSelector}')
+    })`
+    return eval(strFn)
```

Comment:
No need to use eval here .. just pass in `elementSelector` and `qsAll` as execute parameters.

Reply:
Sorry, I'm not following how that would look. Can you give me an example?

Reply:
uff .. ok just saw that you are using the fn selector which can't take arguments like execute. Never mind then